syntax/abstract.mjs

```diff
@@ -10,6 +10,16 @@ import {always, never} from "../lib/combinators.mjs";
 
 export function list_into(Type) {
     switch (Type) {
+        case FTL.AttributeExpression:
+            return ([ref, name]) => {
+                let ref_is_valid =
+                    ref instanceof FTL.MessageReference
+                    || ref instanceof FTL.TermReference;
+                if (ref_is_valid) {
+                    return always(new Type(ref, name));
+                }
+                return never(`Invalid ref type: ${ref.type}.`);
+            };
```

Comment:
I think this part is not needed with the reduced changes after the rebase? Attributes are always on messages or terms still.

Reply:
You're right, but I'd like to keep this here for the sake of completeness. I think it complements the validation of `CallExpression`. Let's keep it for now and I'll see if I can remove it in a follow-up.